packages/terra-tag/src/Tag.jsx

```diff
@@ -97,16 +116,19 @@ class Tag extends React.Component {
     const isClickable = href || onClick;
     const clickableComponent = href ? 'a' : 'button';
     const ComponentType = isClickable ? clickableComponent : 'span';
+    const tagRole = clickableComponent === 'a' ? 'link' : 'button';
 
     return (
       <ComponentType
         {...customProps}
         className={tagClasses}
+        role={tagRole}
         onKeyUp={this.handleKeyUp}
         onBlur={this.handleOnBlur}
-        onClick={onClick}
+        onClick={this.handleClick}
         onFocus={onFocus}
         href={href}
+        data-terra-tag
```

Comment:
Is this a typo? This does not look like a JavaScript variable.

Reply:
Added for selectors here
<img width="710" alt="Screenshot 2023-09-25 at 7 07 49 PM" src="https://github.com/cerner/terra-core/assets/132886427/993b9c8b-e404-48f7-b4ec-ec87309751b7">